dvc/external_repo.py

```diff
@@ -165,11 +313,37 @@ def _add_upstream(self, src_repo):
         self.config["remote"]["auto-generated-upstream"] = {"url": cache_dir}
         self.config["core"]["remote"] = "auto-generated-upstream"
 
+    def get_checksum(self, path):
```

Comment:
Hm, maybe it would make sense to make this part of the tree? Kinda like mtime, size, checksum(or rather hash) also sounds related. Kinda like some fs provide checksums.

Reply:
hmm, I'm not sure about making this a part of the tree, since get_checksum is really a local cache operation and not a tree operation?

I added this function to ExternalRepo before the context managers for using the correct cache & trees, but now that we have the contextmanagers to use the right trees, I think we can just drop this function from ExternalRepo entirely. Originally this was meant to replace a `self.repo.cache.local.get_checksum(...)` call in `RepoDependency._get_checksum`, since getting checksums for git objects depends on the erepo git tree that the DependencyRepo cache is unaware of.  But now we can just use
```
cache = self.repo.cache.local
with cache.erepo_tree(erepo.tree):
    cache.get_checksum(...)
```
directly in DependencyRepo.

Reply:
@pmrowla Hm, I thought we would make it accept obj so that we can compute the checksum that we need directly from git index, without us having to checkout stuff.

Reply:
@efiop we do compute the checksum we need directly from git index (without `git checkout`) though

in LocalRemote we eventually call:
```py
    def get_file_checksum(self, path_info):
        if self.tree.exists(path_info):
            return file_md5(path_info)[0]
        if self._erepo_tree and self._erepo_tree.exists(path_info):
            return file_md5(path_info, self._erepo_tree)[0]
        return file_md5(path_info, self.repo.tree)[0]
```
and in this case, the only place `path_info` will exist is `_erepo_tree`, which is the git-index only GitTree. So `file_md5` will open the fobj and read directly from the GitTree.